File: src/GitOpen.Domain/Status/WorkingFileEntry.cs
namespace GitOpen.Domain.Status;

public enum WorkingFileState { Unmodified, Added, Modified, Deleted, Renamed, Conflicted, Untracked, Ignored }

public sealed record WorkingFileEntry(
    string Path,
    WorkingFileState IndexState,
    WorkingFileState WorkingTreeState,
    string? OldPath = null);

File: lib/domain/status/working_file_entry.dart
import 'package:equatable/equatable.dart';

enum WorkingFileState {
  unmodified,
  added,
  modified,
  deleted,
  renamed,
  conflicted,
  untracked,
  ignored,
}

final class WorkingFileEntry extends Equatable {

  const WorkingFileEntry({
    required this.path,
    required this.indexState,
    required this.workingTreeState,
    this.oldPath,
  });
  final String path;
  final WorkingFileState indexState;
  final WorkingFileState workingTreeState;
  final String? oldPath;

  @override
  List<Object?> get props => [path, indexState, workingTreeState, oldPath];
}
